sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala

```diff
@@ -85,8 +83,9 @@ class DiscoveryImpl(
 
     // possibly filtered or hidden env, passed along for substitution in descriptor options
     val env: Map[String, String] = system.settings.config.getAnyRef("kalix.discovery.pass-along-env-allow") match {
-      case false => Map.empty
-      case true  => sys.env
+      // FIXME: how to do this better?
+      case obj if !obj.asInstanceOf[Boolean] => Map.empty
+      case obj if obj.asInstanceOf[Boolean]  => sys.env
```

Comment:
This is indeed weird. 

What about this? 

```suggestion
      case bool: Boolean => if (bool) sys.env else Map.empty
```

It still feels awkward in pattern matching, but more concise.